### PR TITLE
fix(677) enable users to enter text into drop down boxes

### PR DESCRIPTION
**Pull request by tplevko**
fix: https://github.com/KaotoIO/kaoto-next/issues/677
fix: https://github.com/KaotoIO/kaoto-next/issues/512

With the [typeahead with create](https://www.patternfly.org/components/menus/select/react/typeahead-with-create-option/) 

https://github.com/KaotoIO/kaoto-next/assets/4180208/f6d19c26-0cb6-413e-9258-430f4898f8a0

### CODE REVIEW

File: packages/ui/src/components/Form/customField/TypeaheadField.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        setInputValue(value as string);
+        setFilterValue('');
+        setSelected(value as string);
+        props.onChange(value as string);
+      }
+    }
+    setIsOpen(false);
+    setFocusedItemIndex(null);
+    setActiveItem(null);
+  };
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/TypeaheadField.tsx:101]: [curious-question]: Wouldn't this nested `if` be simplified as:

```typescript
    if (value === 'create') {
        if (!allOptions.some((item) => item.value === filterValue)) {
          allOptions = [...allOptions, { value: filterValue, children: filterValue }];
        }
        setSelected(filterValue);
        setOnCreation(!onCreation);
        props.onChange(filterValue);
        setFilterValue('');
      } else {
        setInputValue(value as string);
        setFilterValue('');
        setSelected(value as string);
        props.onChange(value as string);
      }
    }
    setIsOpen(false);
    setFocusedItemIndex(null);
    setActiveItem(null);
  };
```

File: packages/ui/src/components/Form/customField/TypeaheadField.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+          />
+        ))}
+      </SelectList>
+    </Select>,
+  );
+}
+
+TypeaheadField.defaultProps = {};
+
+export default connectField(TypeaheadField);
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/TypeaheadField.tsx:239]: Could export this component with a name?
image

As an example, like we do for the `DisabledField`
```tsx
export const DisabledField = connectField((props: ConnectedFieldProps<CustomStepsFieldProps>) => {
  return (
    <Card>
      <CardTitle>{props.label}</CardTitle>
      <CardBody>
        <ExpandableDetails details={props}>
          <p>Configuring this field is not yet supported</p>
        </ExpandableDetails>
      </CardBody>
    </Card>
  );
});
```

**Review by lordrip — APPROVED** [review]
Looks great, just a couple 💅 , thanks a lot @tplevko

File: packages/ui/src/components/Form/customField/TypeaheadField.tsx (hunk 1 above)
Comment (tplevko) [reply packages/ui/src/components/Form/customField/TypeaheadField.tsx]: yes, you're right, the first check was redundant. Updated.

File: packages/ui/src/components/Form/customField/TypeaheadField.tsx (hunk 2 above)
Comment (tplevko) [reply packages/ui/src/components/Form/customField/TypeaheadField.tsx]: Sure, updated to export with a name.

**Review by lordrip — APPROVED** [review]

**Pull request merged**